docs/apache-airflow/plugins.rst

```diff
@@ -268,6 +285,7 @@ definitions in Airflow.
         operator_extra_links = [
             S3LogLink(),
         ]
+        listeners = [PrintingRunningListener]
```

Comment:
I'm not sure about the need for the class here. Is there a reason we don't follow the "pluggy" way here and do:

```suggestion
        listeners = [on_task_instance_running]
```

Reply:
The disadvantage of using a class is it might imply that you can store state in the instance between plugin invocations (which isn't the case, not in general terms anyway) where as a function might make that clearer.

Reply:
Generally, in pluggy, we need to register `plugin` that's a namespace that consists of possibly multiple hookimpls, not single functions. 
By passing just a list of functions, we'd later need to construct "synthetic" namespace for it to register.

I think requiring a class, or, possibly a python module, is a good idea - this is what pluggy docs present:

- class https://pluggy.readthedocs.io/en/stable/index.html#a-toy-example
- module https://pluggy.readthedocs.io/en/stable/index.html#the-plugin

Now, it does not need to look like 
```python
        listeners = [PrintingRunningListener]
```

We could require plugin authors to directly implement hookimpls in plugins, for example

```python
class SomeListenerPlugin(AirflowPlugin):
    name = 'some-plugin'
    @hookimpl
    def on_task_instance_running(...):
        print("running")
``` 

or, we could require it to add single pluggy plugin - as class or module - if for some reason we don't want them to implement multiple listener functions in single plugin.

Note, this is separate discussion from https://github.com/apache/airflow/pull/20443#discussion_r778047132

Reply:
Oh yes, you are right. Thinking... 🤔

Reply:
So looking at pytest itself (which is where pluggy came out of) the plugins are def just modules, not classes https://docs.pytest.org/en/6.2.x/writing_plugins.html?highlight=plugin

Reply:
@ashb added check for modules in `add_listener.py` and refactored tests from classes to modules.